packages/typescript-eslint/src/config-helper.ts

```diff
@@ -94,6 +94,7 @@ export function config(
     const extension = {
       ...(config.files && { files: config.files }),
       ...(config.ignores && { ignores: config.ignores }),
+      ...(config.name && { name: config.name }),
```

Comment:
I think we need to be a little smarter here and merge the names to help make things more understandable.

For example -- instead of _overriding_ the name, do like

```ts
name = (() => {
  if (extension.name) {
    if (conf.name) {
      return `${extension.name}__${conf.name}`;
    }
    return extension.name;
  }
  return undefined;
})(); 
```

WDYT?